broker-daemon/relayer/market-watcher.js

```diff
@@ -0,0 +1,132 @@
+const EventEmitter = require('events')
+const { MarketEvent } = require('../models')
+const { migrateStore } = require('../utils')
+
+/**
+ * @class Watch a relayer market and put the events into a data store
+ * It emits two events:
+ * - `sync` when it is caught up to the market
+ * - `end` when it is no longer connected to the market
+ */
+class MarketWatcher extends EventEmitter {
+  /**
+   * Set up the watcher
+   * @param  {grpc.ServerStreaming} watcher        Stream of events from the Relayer
+   * @param  {Sublevel}             store          Leveldb compatible store
+   * @param  {Object}               RESPONSE_TYPES Response types from the proto file
+   * @param  {Object} logger
+   * @return {MarketWatcher}
+   */
+  constructor (watcher, store, RESPONSE_TYPES, logger) {
+    super()
+    this.watcher = watcher
+    this.store = store
+    this.migrating = null
+    this.logger = logger
+    this.RESPONSE_TYPES = RESPONSE_TYPES
```

Comment:
I am confused as to why you are passing this in, when we have hard coded expected response types lower in the file?

Reply:
They aren't hardcoded lower in the file, we're pulling them out of the proto file in relayer-client.js

Reply:
@treygriffith when we upgrade the grpc on broker, we will no longer have access to those enums from the protofile. As a preventative measure, I would hardcode them somewhere

Reply:
NVM, we can hold off until the actual migration. per our talk @treygriffith